src/platform/bouffalolab/BL602/BUILD.gn

```diff
@@ -30,34 +32,43 @@ static_library("BL602") {
   ]
 
   common_sources = [
-    "${chip_root}/src/credentials/CHIPCert.h",
-    "${chip_root}/src/credentials/DeviceAttestationCredsProvider.h",
     "../../FreeRTOS/SystemTimeSupport.cpp",
     "../../SingletonConfigurationManager.cpp",
     "../common/BLConfig.cpp",
     "../common/ConfigurationManagerImpl.cpp",
     "../common/ConnectivityManagerImpl.cpp",
     "../common/DiagnosticDataProviderImpl.cpp",
-    "../common/FactoryDataProvider.cpp",
-    "../common/FactoryDataProvider.h",
     "../common/KeyValueStoreManagerImpl.cpp",
     "../common/Logging.cpp",
     "../common/PlatformManagerImpl.cpp",
   ]
 
   if (chip_enable_ota_requestor) {
-    sources += [
+    common_sources += [
       "../common/OTAImageProcessorImpl.cpp",
       "../common/OTAImageProcessorImpl.h",
     ]
   }
 
   if (chip_enable_ble) {
-    sources += [
+    common_sources += [
       "../common/BLEManagerImpl.cpp",
       "../common/BLEManagerImpl.h",
     ]
   }
+
+  if (chip_enable_factory_data || chip_enable_factory_data_test) {
+    if (chip_enable_factory_data) {
+      defines = [ "CONFIG_BOUFFALOLAB_FACTORY_DATA_ENABLE" ]
```

Comment:
- The code with CONFIG_BOUFFALOLAB_FACTORY_DATA_ENABLE protected in header is unused and now removed.
    ```c++
    private:
    #if !CONFIG_BOUFFALOLAB_FACTORY_DATA_ENABLE
        uint8_t mSpake2pSalt[32];
        uint32_t mSpake2pSaltLen;
        uint8_t mSpake2pVerifier[100];
        uint32_t mSpake2pVerifierLen;
    #endif
    ```

- change to `CONFIG_BOUFFALOLAB_FACTORY_DATA_ENABLE=1` in `.gn`

Reply:
And also add default definition of `CONFIG_BOUFFALOLAB_FACTORY_DATA_ENABLE` in `CHIPDevicePlatformConfig.h`; let other build system knows it can be re-defined.